website/next.config.js

```diff
@@ -14,23 +14,20 @@ const withNextra = nextra({
   flexsearch: false,
   codeHighlight: false,
   defaultShowCopyCode: false,
-  transformPageOpts(pageOpts) {
-    if (pageOpts.filePath === 'pages/_app.mdx') {
-      pageOpts.pageMap = pageOpts.pageMap
-        // remove redundant dynamic page folder
-        .filter((pageItem) => pageItem.name !== '[locale]')
-        .map((pageItem) => {
-          if (pageItem.kind === 'Folder' && pageItem.name.length === 2) {
-            // add missing dynamic index page
-            return {
-              ...pageItem,
-              children: [{ kind: 'MdxPage', name: 'index', route: pageItem.route }, ...pageItem.children],
-            }
-          }
-          return pageItem
-        })
+  transform(result, { route }) {
+    if (route && !result.includes('getStaticProps')) {
+      const banner = `
+import { getPageMap } from '@/components/get-page-map'
+
+export const getStaticProps = async context => ({
+  props: {
+    __nextra_pageMap: await getPageMap('${route.split('/')[1]}')
+  }
+})`
+      result += banner
```

Comment:
What does `banner` mean here?

Reply:
the piece of code to include `getStaticProps` for pages that don't have it